core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -599,22 +599,7 @@ public static void writeLevelCardinalityFile(String loadFolderLoc, String tableN
    * @return
    */
   public static String unescapeChar(String parseStr) {
-    switch (parseStr) {
-      case "\\001":
-        return "\001";
-      case "\\t":
-        return "\t";
-      case "\\r":
-        return "\r";
-      case "\\b":
-        return "\b";
-      case "\\f":
-        return "\f";
-      case "\\n":
-        return "\n";
-      default:
-        return parseStr;
-    }
+    return scala.StringContext.treatEscapes(parseStr);
```

Comment:
I found the functionality of `treatEscapes` is as following. 
```
  /** Expands standard Scala escape sequences in a string.
   *  Escape sequences are:
   *   control: `\b`, `\t`, `\n`, `\f`, `\r`
   *   escape:  `\\`, `\"`, `\'`
   *   octal:   `\d` `\dd` `\ddd` where `d` is an octal digit between `0` and `7`.
   *
   *  @param  str  A string that may contain escape sequences
   *  @return The string with all escape sequences expanded.
   */
```
Do you want to support octal string like \1, \2?
Adding this functionality is good, my concern is that can you find a java function instead of using scala library? Because generaling speaking, core should not dependent on scala library, they should be pure java implementation. 

Another thing is can you add a testcase for this PR?